### DISCUSSION TITLE
Can I rename the table name all uppercase? MySQL

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [prisma/prisma#23489] [Q&A]
### Question

Is there a way that Prisma generates table names in uppercase without affecting its proper behavior? Currently, the `@@map("AIKON_FAMILIA")` and `@@map("AIKON_ARTICULO")` generates tables "aikon_familia" and "aikon_articulo" in MySQL. I'd like them all in uppercase. How could I do that?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model AIKON_FAMILIA {
  FA_CODIGO       String  @db.VarChar(20) @id
  FA_NOMBRE       String? @db.VarChar(30)
  FA_NIVEL        String? @db.VarChar(2)
  FA_PALM         String? @db.VarChar(1)

  ARTICULO        AIKON_ARTICULO[]
  @@map("AIKON_FAMILIA")
}

model AIKON_ARTICULO {
  AR_CODIGO       String @db.VarChar(24) @id
  AR_DESCRI       String? @db.VarChar(200)
  UM_CODIGO       String? @db.VarChar(2)
  AR_MEMO         String? @db.VarChar(4000)
  II_CODIGO       String? @db.VarChar(2)
  AR_BARRAS       String? @db.VarChar(20)
  AR_COSNET       Decimal? @db.Decimal(15,2)
  MA_CODIGO       String? @db.VarChar(3)
  RE1_CODIGO      String? @db.VarChar(5)
  RE2_CODIGO      String? @db.VarChar(5)
  ALI_CODVEN      String? @db.VarChar(2)
  ALI_CODCOM      String? @db.VarChar(2)
  ESA_CODIGO      String? @db.VarChar(2)
  AR_ALTO         Float?
  AR_ANCHO        Float?
  AR_PROFUNDO     Float?
  AR_COLOR        String? @db.VarChar(20)
  AR_DESMAX       Decimal? @db.Decimal(5,2)
  AR_DESCRIA      String? @db.VarChar(200)
  AR_COSREP       Decimal? @db.Decimal(15,2)
  AR_COSUCP       Decimal? @db.Decimal(15,2)
  AR_FECHAMODIF   DateTime?
  PR_CODIGO       String? @db.VarChar(5)
  AR_MESESGARANTIA Float?

  AIKON_FAMILIA   AIKON_FAMILIA @relation(fields:[FA_CODIGO], references:[FA_CODIGO])
  FA_CODIGO       String @db.VarChar(20)
  
  @@map("AIKON_ARTICULO")
}
```
- OS: Windows
- Database: MySQL
- Node.js version: 18.14.1

- prisma: 5.9.1
- @prisma/client: 5.9.1

**Comment by janpio** [comment]
Your MySQL seems to be configured to create tables lowercase no matter the SQL that is being executed. You will need to change that configuration to honor the actual capitalization. (Note that this might be problematic on Windows, I think I had problems around that in the past.)

(Take a look at the generated `migration.sql` in the `migrations` folder to see the SQL that Prisma generated and just executes.)

**Reply by lucasdavidferrero** [reply]
Inspecting the migration file, I see that Prisma generates `CREATE TABLE` with uppercase names. Seems that MySQL on Windows is configured that way. Thanks for pointing out that it may be problematic enabling uppercase tables names in MySQL with Windows. I'll keep thing safe and leave it as default.

**Reply by janpio** [reply]
Then I would recommend to make sure that the table names in your Prisma schema are the same as the ones that are created by MySQL. If you ever want to migrate to another OS (or just deploy on Linux for example), you will otherwise run into trouble. It's a weird thing that MySQL does there.
